File: serialization/ruby/json/microsoft_kiota_serialization/lib/microsoft_kiota_serialization/json_parse_node_factory.rb
require 'microsoft_kiota_abstractions'
require 'json'

module MicrosoftKiotaSerialization
  class JsonParseNodeFactory
    include MicrosoftKiotaAbstractions::ParseNodeFactory

    def get_valid_content_type
      'application/json'
    end

    def get_parse_node(content_type, content)
      if !content_type
        raise StandardError, 'content type cannot be undefined or empty'
      elsif get_valid_content_type != content_type
        raise StandardError, `expected a #{get_valid_content_type} content type`
      end
      raise StandardError, 'content cannot be undefined or empty' unless content

      JsonParseNode.new(JSON.parse(content))
    end
  end
end

File: serialization/typescript/json/src/browser/jsonParseNodeFactory.ts
import { ParseNode, ParseNodeFactory } from "@microsoft/kiota-abstractions";
import { JsonParseNode } from "../jsonParseNode";

export class JsonParseNodeFactory implements ParseNodeFactory {
    public getValidContentType() : string { return "application/json"; }
    public getRootParseNode(contentType: string, content: ArrayBuffer): ParseNode {
        if(!contentType) {
            throw new Error("content type cannot be undefined or empty");
        } else if (this.getValidContentType() !== contentType) {
            throw new Error(`expected a ${this.getValidContentType()} content type`);
        }
        if(!content) {
            throw new Error("content cannot be undefined of empty");
        }
        const contentAsJson = this.convertToJson(content)
        return new JsonParseNode(contentAsJson);
    }

    private convertToJson(content:ArrayBuffer){
        // Maintaining private function to unit test convertToJson
        const decoder = new TextDecoder();
        const contentAsStr = decoder.decode(content);
        const contentAsJson = JSON.parse(contentAsStr);
        return contentAsJson;
    }
}